php-checks/src/main/java/org/sonar/php/checks/ConditionalOnNewLineCheck.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * SonarQube PHP Plugin
+ * Copyright (C) 2010-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.php.checks;
+
+import java.util.Arrays;
+import java.util.List;
+import org.sonar.check.Rule;
+import org.sonar.plugins.php.api.tree.Tree;
+import org.sonar.plugins.php.api.tree.lexical.SyntaxToken;
+import org.sonar.plugins.php.api.tree.statement.BlockTree;
+import org.sonar.plugins.php.api.tree.statement.ElseClauseTree;
+import org.sonar.plugins.php.api.tree.statement.ElseifClauseTree;
+import org.sonar.plugins.php.api.tree.statement.IfStatementTree;
+import org.sonar.plugins.php.api.tree.statement.StatementTree;
+import org.sonar.plugins.php.api.visitors.PHPSubscriptionCheck;
+
+@Rule(key = ConditionalOnNewLineCheck.KEY)
+public class ConditionalOnNewLineCheck extends PHPSubscriptionCheck {
+
+  public static final String KEY = "S3972";
+
+  private SyntaxToken previousClosingBracketToken;
+
+  @Override
+  public List<Tree.Kind> nodesToVisit() {
+    return Arrays.asList(Tree.Kind.COMPILATION_UNIT, Tree.Kind.IF_STATEMENT, Tree.Kind.ELSEIF_CLAUSE, Tree.Kind.ELSE_CLAUSE);
+  }
+
+  @Override
+  public void visitNode(Tree tree) {
+    if (tree.is(Tree.Kind.COMPILATION_UNIT)) {
+      previousClosingBracketToken = null;
+      return;
+    }
+    if (tree.is(Tree.Kind.IF_STATEMENT) && previousClosingBracketToken != null) {
+      SyntaxToken ifToken = ((IfStatementTree) tree).ifToken();
+      if (ifToken.line() == previousClosingBracketToken.line()) {
```

Comment:
Maybe we can exclude cases like:
```
if ($x) { foo(); } if ($y) { foo(); }
```
(which is a corner case)

Reply:
Ignored by discarding single-line blocks when setting the previous token.